navigation/AuthLoading.js

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { ActivityIndicator, View, Text } from 'react-native';
+import * as SecureStore from 'expo-secure-store';
+
+export default class AuthLoadingScreen extends React.Component {
+  constructor(props) {
+    super(props);
+    this._bootstrapAsync();
+    state = {
+      authObject: null
+    }
+  }
+
+  _bootstrapAsync = async () => {
+    const authObject = await SecureStore.getItemAsync('authObject');
+
+    this.props.navigation.navigate(authObject ? 'Main' : 'Auth');
+    // this.props.navigation.navigate('Auth');
+  };
+
+  render() {
```

Comment:
If it takes a few milliseconds for the user's authentication to resolve with the backend then it may be better to show a blank screen then a flash of the activity indicator as it may be disorienting to the user. This can be achieved by `return null;` in the component while auth is resolving.